pkg/build/api/v1/conversion.go

```diff
@@ -166,17 +166,27 @@ func convert_api_BuildSource_To_v1_BuildSource(in *newer.BuildSource, out *Build
 	if err := s.DefaultConvert(in, out, conversion.IgnoreMissingFields); err != nil {
 		return err
 	}
+	typeFound := false
 	switch {
+	case len(in.Images) > 0:
+		out.Type = BuildSourceImage
+		typeFound = true
```

Comment:
the way it is now, any build w/ images>0 will be labeled as BuildSourceImage.

you want this to be the last check in the case.  (basically the cases be listed in priority order: Git, Binary, Dockerfile, SourceImage)

you can also make the out.Type=BuildSourceNone be the default case for the switch, rather than a separate if.